src/cargo/sources/git/utils.rs

```diff
@@ -453,17 +455,14 @@ impl<'a> GitCheckout<'a> {
                 }
                 new_parent_remote_url.set_path(&new_path);
 
-                match new_parent_remote_url.join(child_url_str) {
-                    Ok(x) => x.to_string(),
-                    Err(err) => Err(err).with_context(|| {
-                        format!(
-                            "failed to parse relative child submodule url `{}` using parent base url `{}`",
-                            child_url_str, new_parent_remote_url
-                        )
-                    })?,
-                }
+                new_parent_remote_url.join(child_url_str).with_context(|| {
+                    format!(
+                        "failed to parse relative child submodule url `{}` using parent base url `{}`",
+                        child_url_str, new_parent_remote_url
```

Comment:
Not blockers

* use implicit arg capture maybe?
* perhaps wrap the line for readability in source code level

```suggestion
                        "failed to parse relative child submodule url `{child_url_str}` \
                        using parent base url `{new_parent_remote_url}`"
```